server/users.ts

```diff
@@ -1018,20 +1019,20 @@ export class User extends Chat.MessageContext {
 	/**
 	 * @param updated the settings which have been updated or none for all settings.
 	 */
-	getUpdateuserText(...updated: string[]) {
+	getUpdateuserText(...updated: UserSetting[]) {
 		const named = this.named ? 1 : 0;
-		const diff = {};
+		const diff: Partial<Pick<User, UserSetting>> = {};
```

Comment:
tbh I would just let `diff` be `any`. It's about to become JSON anyway, and building JSON usually requires things TS doesn't understand.